fdbserver/workloads/TenantManagementWorkload.actor.cpp

```diff
@@ -1512,6 +1512,18 @@ struct TenantManagementWorkload : TestWorkload {
 			hasSystemTenantGroup = hasSystemTenantGroup || newTenantGroup.orDefault(""_sr).startsWith("\xff"_sr);
 			configuration["tenant_group"_sr] = newTenantGroup;
 		}
+		// Configuring 'assignedCluster' requires reading existing tenant entry which is handled in
+		// `updateManagementCluster`.
+		state bool assignToDifferentCluster = false;
+		if (operationType == OperationType::METACLUSTER && (!hasInvalidOption || deterministicRandom()->coinflip())) {
```

Comment:
The previous use of `!hasInvalidOption` was to make sure we did some configuration. In other words, we either used an invalid option or we set the tenant group.

Now, we have three candidates, so I might structure this a little bit differently to try all combinations.

One option is to coinflip both tenant group and assigned cluster, and then set an invalid option if there is no other configuration set or by coinflip. Something like this:

```
if (coinflip()) { setTenantGroup(); }
if (coinflip()) { setAsssignedCluster(); }
if (configuration.empty() || coinflip()) { setInvalidOption(); }
```

Reply:
Maybe I am missing something. When looking at the code, if `operationType == SPECIAL_KEYS`,  how does `ConfigureTenantImpl` throw if the tenant does not exist?

Reply:
Also just realize that in the case of SPECIAL_KEYS, we need to make sure `configuration` is not empty.